Org.americanmonk.tipitaka pali reader

bksubhuti opened the pull request:
I have included an xml and desktopfile on the root with the yml file.. Please know that these files are in the tar.gz 
However, I used the appstream metaInfo Creator tool to make the newer xml file.  It is slightly different.

### Please confirm your submission meets all the criteria

- [ x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [ x] My pull request follows the instructions at [App Submission][submission].
- [ x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [ x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

bksubhuti commented:
did I need to reopen the old one and submit.. i made a mistake by closing the previous submission and trying again.  I read I should try to keep the same one open.  It would be good if you can give more detailed instructions if / when refusing.

bksubhuti commented:
I tried to search for the app but I didn't see it.  I searched for tipitaka  and I didn't get anything.

bksubhuti commented:
I looked some more on the appstream validation thing.. It is totally against the model I used and copied.. and says it does not work .. as you might  be  saying  too..  In fact, i have no clue how this works or what to do.. I just took the app that was called feeel that was verified as working and changed the parameters for a binary flutter submission as there are no sufficient documentation for flutter builds (there is plenty of good documents on how to submit a 2 line bash file).... 
I'm not sure what you want me to do.  It would be good if there was a model I could follow.  The docs are not suitable for someone like myself to figure out.   I'd be happy to make a video to help.. flutter is the most popular  multi platform for development and could be the savior for Linux App eco system.   

I currently ship Appimages which are in trouble of survival because of fuse3 crashing Ubuntu systems.  They will soon static link their own spliced fuse3 library.  It would be good to make a flatpak, but It is taking just way too long and it seems more geared for c c++ devs who want to link to flathub libraries.

razzeee commented on org.americanmonk.TipitakaPaliReader.xml:
```diff
@@ -0,0 +1,32 @@
+  <id>org.americanmonk.TipitakaPaliReader</id>
+  
+  <name>Tipitaka Pali Reader</name>
+  <summary>This is an app to read the Theravāda Buddhist Tipitaka Pāḷi texts.  It allows one to click on words and get definitions effortlessly.  </summary>
```
Ideally your summary should be shorter. This is not a requirement, but just a tip.

It also shouldn't end with a fullstop or spaces :)

bksubhuti replied:
I can change it.. but is that why my PR is not approved?  I'll have to figure out how to do a pr update.  But as you said.. not required.

bksubhuti replied:
I should mention that the xml was just checked into the root to satisfy some type of requirement.  It is actually bundled into the gz located in the binary source download.  I highly doubt that changing the xml that I submitted to satisfy this requirement will actually affect anything.  I'm not sure how to do packaging with an external desktop and xml.  Is there an example on how to do that.. like someone who was accepted and did a binary submission?  TBH.. the bundling of all the stuff in a single directory threw me off for some time.  I'd also like to do that without the directories but I don't have any sample files.  If I had a sample.. i could duplicate it.  Please include a sample.

for the mean time.. i'll push an edit to the root xml in this PR, but not in the gz binary bundle.

bksubhuti commented:
This is what I mean about my directory structure .. if you unzip my gz file listed in the yml you will see this image below:  So this is why a root level xml file seems redundant.  I highly doubt that the flat pak will take the root level xml that I submitted.. why?  Because I have installed without it .. also "feeel" which is an accepted app on flathub does not do this either.. we both have the same gz directory structure and similar yml recipes.

![image](https://github.com/flathub/flathub/assets/18184280/de735894-f103-4e84-8d90-d32f56fd9475)


and this is the bin dir:
![image](https://github.com/flathub/flathub/assets/18184280/fecb8671-22dd-44f2-8fe4-8f61cd8aa105)

razzeee commented:
Might have been an oversight from @hfiguiere then?

bksubhuti commented:
If you look at my original PR or even the top of this one.. I say so.  However, perhaps a picture is worth a thousand words.  The idea is to make a release find out the specification and then make an update to the xml file to be shorter.  aka first run.  I can do another directory structure and have the xml at the root.. but this is what I have.. If you show me a project that is a binary release that has an acceptable structure (flathub standards).. as I have said many times over.. I'm willing to do that.

The only comment I got was it does not have the appstream file.  

As I said before, I'm willing to make a video.  When i have trouble accomplishing something I sometimes make a video (if I'm finally successful).  It is not to be popular.. but to help people.
Here is the channel.. only a few vids.
https://www.youtube.com/@technologyfrombhante5764/videos

razzeee commented:
You need to chill down. This is all done on the sides, most people here are volunteers. So nobody is looking at this 9 to 5 and getting through them. There are also mere requests then yours, so  response times might vary due to that.

bksubhuti commented:
sure no problem.. thanks for taking a look.

cxrvh commented:
You have to include the appstream file at a specific location so that Flatpak can find it. See:
[Requirements & Conventions - AppData files](https://docs.flatpak.org/en/latest/conventions.html#appdata-files)
[AppData guidlines - Path and filename](https://docs.flathub.org/docs/for-app-authors/appdata-guidelines#path-and-filename)

You can just take it, like feeel, from your archive, you don't need a copy in your Flathub repository. But you should make sure your AppData file is valid in both cases.

If you add any files, like a Desktop or AppData file, to your Flathub repository, then you have to use these files in your Flatpak manifest, too. They don't get automatically used. I think the only exception is the `flathub.json`.

bksubhuti commented:
Please know that my original submission was exactly that.  I did install on my machine. I first did locally as source file , then from a url download.  Everything was fine.  However, The first reviewer just kept saying no appstream file .  I mentioned the project I followed .. SO just to satisfy the requirements I added to the root.

So he was running the app stream utility.. and i recently read that document and the utility does not validate 
that the appstream is proper.  However, the app stream is proper.  There is virtually no difference.

I'm just baffled that FEEEL was able to get approved and mine was not.  We have virtually the same project.
The utility is looking for xml file.. but that is not there.  IT is in my gz file on the internet.  So of course it won't work.  This is very frustrating.  It seems like the best thing to do is to start over again and copy the structure of the documentation.  

I don't do well with this style of documentation.  Would you be so kind (I have asked this maybe more than 5 times) would you be so kind to direct me to a successfully submitted flutter app that I can copy. ?  Are you telling me that besides the "feeel" guy, I'm the second person to submit a flutter app?

hfiguiere commented:
> Please know that my original submission was exactly that.

Not it wasn't. The appstream file was NEVER installed.

bksubhuti commented:
This was the original PR.  I didn't know I should just edit the PR.  I have not done this before so I started a new one.

https://github.com/flathub/flathub/pull/4299#discussion_r1248857559

The archive was never changed.  I have yet to upload a new url.. simple wget the archive and see.
```

      - type: archive
        url: https://github.com/bksubhuti/tipitaka-pali-reader/releases/download/v2.3.4-52/tpr.tar.gz
        sha256: 0260306a2102a99acd6add4901798d9861d63ef565464c39784b958f6f80ccfd
        dest: "tpr"
```

cxrvh commented:
You install the desktop file & icons but don't do anything with the appstream/appdata/metainfo file.

Just `install` the appstream file to the correct location with the **correct name** (see the links in my previous answer) & everything should be fine.

It's not enough that it's in your archive / folder. It has to be at the correct location!

I can't help with any other flutter app. But any other binary app should be similar, as an example (one of the first in the current repository list): https://github.com/flathub/dev.aunetx.deezer/blob/master/dev.aunetx.deezer.yml

bksubhuti commented:
I'll give it a try.. restructure my app.  

I don't understand the docs.. .  
I'm assuming i'm missing this (because i now have an example besides a helloworld.sh file.. (which also does not cover xml process etc)

```
- install -Dm644 dev.aunetx.deezer.appdata.xml /app/share/appdata/dev.aunetx.deezer.appdata.xml
      - install -Dm644 /app/main/resources/dev.aunetx.deezer.svg /app/share/icons/hicolor/scalable/apps/dev.aunetx.deezer.svg
      - install -Dm644 /app/main/resources/dev.aunetx.deezer.desktop /app/share/applications/dev.aunetx.deezer.desktop
```

`install -Dm644  `
This is also not in the tutorial.
I also cannot find anything docs about install dm644

However, I'll just emulate the directory structure and xml stuff and see what happens.

Take note:
The documentation searching internally or google search confuses install for developers and install for users.
Without a sample, it is impossible for people to know what to do.  I hope you can give several realworld examples in your tutorial, maybe even make a video.  If it is unusable and a bad experience, people will not bother.

bksubhuti commented:
I cannot get it to work.. I barely got it to work last time.. changing the directory structure kept giving me errors that it cannot find my executable.  I'm at a loss.. this is taking too much of my time.  I have an AppImage.  I'm happy.  Linux users can use my AppImage without installing 2gb of flatpak support files for my 100+ MB app.
I allocated 33 gb for Ubuntu Mate to make this flatpak and kept running out of space.  It is just too heavy and too time consuming.   

Maybe I'll try in a year or two when more people flutter people have been successful, there are videos that explain the process and better documentations for people like myself, or the flatpak builder gives better feedback.. It is too much trouble for me.

I was ready to make a video to help people. but I'm not successful.  Now, I'd like to see the Flatpak team do that.  It will save you time in the long run if you do this.  Surely.. i cannot be the first to complain like this and have trouble.  If I am.. i take full responsibility for my lack of knowhow.  I used to use linux full time so I am sympathetic to linux users.  I still support linux with appimage.  I'll wait until the snap flatpak appimage war finishes its course.. Right now.. AppImage wins.

I will close this PR.

razzeee commented:
Have fun, this should be working if you fix your appdata, as it still seems to point to feeel at places.

https://github.com/razzeee/org.americanmonk.TipitakaPaliReader

(I had a crash on the first start, but that likely was the app itself)

Some other recommendations

1. double check, that you actually own americanmonk.org
2. Improve your desktop files/appdata, summary is bad and so are appnames with `_`

bksubhuti commented:
fixed and uploaded with the old data structure based on @razzeee razzee's help.. it is appreciated.. i totally gave up before.  Let's see what happens now.

razzeee commented:
I can build this fine with your latest changes.

One thing I would like to see changed in your `.desktop` would be `Name=tipitaka_pali_reader`  to `Name=Tipitaka Pali Reader`. At least for me gnome shows the underline version in places. 

To be clear, I don't have anything to do with these reviews, just pointing out stuff that I find.

bksubhuti commented:
yes.. can fix.
Bhante Subhuti

https://AmericanMonk.org
https://withmetta.net



On Mon, 10 Jul 2023 at 15:05, Kolja ***@***.***> wrote:

> I can build this fine with your latest changes.
>
> One thing I would like to see changed in your .desktop would be
> Name=tipitaka_pali_reader to Name=Tipitaka Pali Reader. At least for me
> gnome shows the underline version in places.
>
> To be clear, I don't have anything to do with these reviews, just pointing
> out stuff that I find.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/flathub/flathub/pull/4302#issuecomment-1628591633>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEKXQWE6V6COKNKN75WCBQTXPPEGXANCNFSM6AAAAAAZ3L6VCM>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>

bksubhuti commented:
okay fixed.. the desktop.. i fixed the sha again.

bksubhuti commented:
If you want to have flatpak build flutter projects from source.. I have a build script I use for appimage on Digital Ocean 18.04 droplet.  github automations does not go down to 18.04 anymore and I don't know who to do dockers so I just build by script.
https://github.com/bksubhuti/tipitaka-pali-reader/blob/master/serverbuildcommands.sh

barthalion commented:
bot, build org.americanmonk.TipitakaPaliReader

barthalion commented:
```
builddir/files/share/appdata/org.americanmonk.TipitakaPaliReader.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• tag-missing           : <release> required
• tag-missing           : <url> is not present
```

bksubhuti commented:
I pushed with a new sha256 to cover the new upload
I added the content_rating using the online app to make the tag  .. Please note it is religious texts.. so not all content is for babies even though I said it was okay.. mostly in the context of games it seems.

I added my homepage url for the app. and the github issues page
I added release tag as well.

All of these were modeled based on the online app to make content rating and also 
the deezer yml posted in above comments.  I ran the flatpak-builder after uploading to be sure the sha256 was correct and matched what was uploaded during the release edit on github.


<content_rating> 
<release> 
 <url>

bksubhuti commented:
The previous push was dated before the last few comments (and build) because when i boot linux, it changes my windows date.  Please see previous commit details.. 
I just removed one line so it would push a new commit at the right time stamp.

bksubhuti commented:
I have not heard back in a few weeks.  I'd like to know if this can finally get accepted .
How often am i allowed to send updates (like new updated versions of the product)?

razzeee commented:
Hey, this is mostly run by volunteers and some of those are likely already traveling to one of the upcoming community conferences. So you might need to be a bit more patient.

You can update as often as you want, but I would suggest to not do daily builds, as that would be pretty bad on the flathub infrastructure.

bksubhuti commented:
Okay.. we will be making a release soon with some new improvements.  So if
the conference delays things, I will change the binary and sha256 and
resubmit

We don't publish daily builds
Because pr's are slow, I'll try to upload the most stable period with
important changes or features.  And if they need adjusting while.waiting ,
I'll do that too.

Normally android and msoft releases are more frequent but not "daily"..
we release when we feel things are at a checkpoint.  So the flathub
submissions should be better quality than android.

On Mon, Jul 24, 2023, 14:43 Kolja ***@***.***> wrote:

> Hey, this is mostly run by volunteers and some of those are likely already
> traveling to one of the upcoming community conferences. So you might need
> to be a bit more patient.
>
> You can update as often as you want, but I would suggest to not do daily
> builds, as that would be pretty bad on the flathub infrastructure.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/flathub/flathub/pull/4302#issuecomment-1647525092>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEKXQWCLI7PCX7XZOEYQIOLXRY4E3ANCNFSM6AAAAAAZ3L6VCM>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>

barthalion commented:
/merge

bksubhuti commented:
your time and effort is appreciated.
